File: php/01-exercise/00-script.php
<?php
/**
 * Función para recortar una cadena según el número de caracteres indicados.
 *
 * @param string $cadena La cadena de texto a recortar.
 * @param int $longitud La longitud a la cual recortar la cadena.
 * @return string La cadena recortada.
 */
function cortarCadena($cadena, $longitud) {
    // Verificar que la longitud es un entero positivo
    if (!is_int($longitud) || $longitud < 0) {
        return "Error: La longitud debe ser un entero positivo.";
    }

    // Verificar que la cadena no esté vacía
    if (empty($cadena)) {
        return "Error: La cadena de texto no puede estar vacía.";
    }

    // Recortar la cadena y devolver el resultado
    return substr($cadena, 0, $longitud);
}

// Ejemplo de uso
$cadena = "Hola Mundo";
$longitud = 4;
$resultado = cortarCadena($cadena, $longitud);
echo $resultado; // Salida: Hola
?>

File: python/01-exercise/00-script.py
#!/usr/bin/env python3


# Programa una función que te devuelva el texto recortado según el número de caracteres 
# indicados, pe. miFuncion("Hola Mundo", 4) devolverá "Hola".


def cortar_cadena(cadena,longitud):
    """
    Recorta una cadena por el numero de longitud.

    Args:
        cadena (str): La cadena de texto la cual se recortara.
        longitud (int): El numero de caracteres por los que se recortara la cadena

    Returns:
        str: La cadena recortada.

    Raises:
        TypeError: Si el argumento proporcionado no es una cadena de texto o si
        la longitud no es un entero.
    """
    if not isinstance(cadena, str) or not cadena:
        raise TypeError("El argumento debe ser una cadena de texto.")
    
    if not isinstance(longitud, int) or not longitud:
        raise TypeError("El argumento debe ser un entero.")
    
    try:
        # Utilizamos rabanadas para recortar la cadena
        cadena_recortada = cadena[0:longitud]
        return cadena_recortada
    except Exception as e:
        # Capturamos cualquier excepción inesperada
        print(f"Ocurrió un error: {e}")
        raise

# Ejemplo de uso
try:
    resultado = cortar_cadena("Hola Mundo",4)
    print(f"El número de caracteres es: {resultado}")
except TypeError as e:
    print(e)
except Exception as e:
    print(f"Se produjo un error inesperado: {e}")


# segundo ejemplo de uso
try:
    resultado = cortar_cadena("Hola mundo")
    print(f"El número de caracteres es: {resultado}")
except TypeError as e:
    print(e)
except Exception as e:
    print(f"Se produjo un error inesperado: {e}")